neurow/lib/neurow/public_api/endpoint.ex

```diff
@@ -194,58 +196,84 @@ defmodule Neurow.PublicApi.Endpoint do
     {conn, sent}
   end
 
-  defp loop(conn, sse_timeout, keep_alive, last_message, last_ping, jwt_exp) do
-    receive do
-      {:pubsub_message, message} ->
-        conn = write_data_chunk(conn, message)
-        Stats.inc_msg_published()
-        new_last_message = :os.system_time(:millisecond)
-        conn |> loop(sse_timeout, keep_alive, new_last_message, new_last_message, jwt_exp)
+  def loop(conn, sse_timeout_ms, keep_alive_ms, last_message_ts, last_ping_ts, jwt_exp) do
+    now_ms = :os.system_time(:millisecond)
 
-      # Consume useless messages to avoid memory overflow
-      _ ->
-        conn |> loop(sse_timeout, keep_alive, last_message, last_ping, jwt_exp)
-    after
-      1000 ->
-        now_ms = :os.system_time(:millisecond)
-
-        cond do
-          # SSE Timeout
-          now_ms - last_message > sse_timeout ->
-            Logger.debug("Client disconnected due to inactivity")
-            conn |> write_raw_chunk("event: timeout")
-
-          # SSE Keep alive, send a ping
-          now_ms - last_ping > keep_alive ->
-            conn
-            |> write_raw_chunk("event: ping")
-            |> loop(sse_timeout, keep_alive, last_message, now_ms, jwt_exp)
+    cond do
```

Comment:
I reversed the order of the `receive` block and all the conditions in the `cond` block in the last commit.

If it less readable this way, I can just revert the last commit on the branch.